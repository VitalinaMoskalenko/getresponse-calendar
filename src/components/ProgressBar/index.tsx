import React from "react";
import styled from "styled-components";
import { SmallBody } from "../Paragraphs";

type LinePropsType = {
  isAccomplished?: boolean;
};

type RoundPropsType = {
  isAccomplished?: boolean;
  isCurrent?: boolean;
};

const Container = styled.div``;
const Round = styled.div<RoundPropsType>`
  height: 10px;
  width: 10px;
  border: 2px solid
    ${({ isAccomplished, theme }) =>
      isAccomplished ? `${theme.colors.green}` : `${theme.colors.darkGrey}`};
  border-radius: 6px;
  background-color: ${({ isAccomplished, isCurrent, theme }) =>
    isAccomplished
      ? `${theme.colors.green}`
      : isCurrent
      ? `${theme.colors.white}`
      : `${theme.colors.darkGrey}`};
`;

const Line = styled.div<LinePropsType>`
  margin-top: 6px;
  width: 10px;
  border-bottom: 2px solid
    ${({ isAccomplished, theme }) =>
      isAccomplished ? `${theme.colors.green}` : `${theme.colors.darkGrey}`};
`;

const ProgressItem = styled.div`
  display: flex;
  flex-direction: row;
  align-items: start;
  margin-top: 5px;
`;

const ContainerCountWeek = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

type PropsType = {
  countWeek: number;
  accomplishedWeek?: number;
  progressTitle: string;
};

const ProgressBar = ({
  countWeek,
  accomplishedWeek,
  progressTitle,
}: PropsType) => {
  return (
    <Container>
      <SmallBody>{progressTitle}</SmallBody>
      <ProgressItem>
        {[...Array(countWeek)].map((e, i) => (
          <ProgressItem key={i}>
            <ContainerCountWeek>
              {accomplishedWeek ? (
                <Round
                  isAccomplished={i < accomplishedWeek}
                  isCurrent={i === accomplishedWeek}
                />
              ) : (
                <Round isCurrent={i === 0} />
              )}
              <SmallBody>{i + 1}</SmallBody>
            </ContainerCountWeek>
            {i !== countWeek - 1 && (
              <Line
                isAccomplished={
                  i < (accomplishedWeek ? accomplishedWeek - 1 : -1)
                }
              />
            )}
          </ProgressItem>
        ))}
      </ProgressItem>
    </Container>
  );
};

export default ProgressBar;
