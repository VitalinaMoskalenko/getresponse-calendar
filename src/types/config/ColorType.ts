export enum ColorType {
  white = "white",
  black = "black",
  lightGrey = "#e1e1e1",
  darkGrey = "#535353",
  green = "#b2c200",
  orange = "#ff801a",
}
