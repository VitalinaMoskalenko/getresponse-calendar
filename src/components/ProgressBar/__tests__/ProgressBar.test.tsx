import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import ProgressBar from "../";
import { theme } from "../../../config";

it("renders ProgressBar correctly", () => {
  const progressBarComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ProgressBar countWeek={1} accomplishedWeek={1} progressTitle="" />
      </ThemeProvider>
    )
    .toJSON();

  expect(progressBarComponent).toMatchSnapshot();
});
