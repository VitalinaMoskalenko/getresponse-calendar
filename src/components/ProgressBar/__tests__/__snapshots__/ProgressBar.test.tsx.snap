// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProgressBar correctly 1`] = `
<div
  className="sc-dlnjwi"
>
  <p
    className="sc-gtsrHT fsyTPD"
  >
    
  </p>
  <div
    className="sc-jSFjdj iHQLjq"
  >
    <div
      className="sc-jSFjdj iHQLjq"
    >
      <div
        className="sc-gKAaRy ezgkaX"
      >
        <div
          className="sc-hKFxyN dwjDcy"
        />
        <p
          className="sc-gtsrHT fsyTPD"
        >
          1
        </p>
      </div>
    </div>
  </div>
</div>
`;
